enterprise/backend/src/metabase_enterprise/serialization/v2/extract.clj

```diff
@@ -131,19 +71,39 @@
   (let [card (t2/select-one [Card :collection_id :name] :id card-id)]
     (format "Card %d (%s from collection %s)" card-id (:name card) (collection-label (:collection_id card)))))
 
-(defn- escape-report
-  "Given the analysis map from [[escape-analysis]], report the results in a human-readable format with Card titles etc."
-  [{:keys [escaped-dashcards escaped-questions]}]
-  (when-not (empty? escaped-dashcards)
-    (doseq [[dash-id card-ids] escaped-dashcards
-            :let [dash-name (t2/select-one-fn :name Dashboard :id dash-id)]]
-      (log/warnf "Failed to export Dashboard %d (%s) containing Cards saved outside requested collections: %s"
-                 dash-id dash-name (str/join ", " (map card-label card-ids)))))
-
-  (when-not (empty? escaped-questions)
-    (log/warnf "Failed to export Cards based on questions outside requested collections: %s"
-               (str/join ", " (for [[curated-id alien-id] escaped-questions]
-                                (str (card-label curated-id) " -> " (card-label alien-id)))))))
+(defn- parse-target [[model-name id :as target]]
+  (if (string? id)
+    [model-name (serdes/eid->id model-name id)]
+    target))
+
+(defn- escape-analysis [{colls "Collection" cards "Card" :as _nodes} reasons]
```

Comment:
It's not clear at all to me, looking at the code for both `escape-analysis` and `extract-subtrees`, what `reasons` are meant to be. I don't grok the shape of the `nodes`, which means I can't grok the (fairly obscure) code for `reasons` either.

Reply:
Oh right, I wanted to make it `mu/defn`, but then all the tests were broken and I promptly forgot about this. 😁